docs/content/latest/explore/ysql-language-features/data-manipulation.md

```diff
@@ -0,0 +1,343 @@
+
+You can use automatic timestamps to keep track of when data in a table was added or updated. 
+
+The date of the data creation is typically added via a  `created_at` column with a default value of `NOW()`, as shown in the following example: 
+
+```sql
+CREATE TABLE employees (
+    employee_no integer NOT NULL,
+    name text,
+    department text,
+  	created_at TIMESTAMP NOT NULL DEFAULT NOW()
+);
+```
+
+To track updates, you need to use triggers that let you define functions executed when an update is performed. The following example shows how to create a function in PL/pgSQL that returns an object called `NEW` containing data being modified:
+
+```sql
+CREATE OR REPLACE FUNCTION trigger_timestamp()
+RETURNS TRIGGER AS $$
+BEGIN
+  NEW.updated_at = NOW();
+  RETURN NEW;
+END;
+$$ LANGUAGE plpgsql;
+```
+
+The following examples create a table and connect it with a trigger that executes the `trigger_timestamp` function every time a row is updated in the table:
+
+```sql
+CREATE TABLE employees (
+    employee_no integer NOT NULL,
+    name text,
+    department text,
+  	updated_at TIMESTAMP NOT NULL DEFAULT NOW()
+)
+```
+
+```sql
+CREATE TRIGGER set_timestamp
+BEFORE UPDATE ON employees
+FOR EACH ROW
+EXECUTE PROCEDURE trigger_timestamp();
+```
+
+## Defining NOT NULL Constraint
+
+YSQL lets you define various constraints on columns. One of these constrains is `NOT NULL`. You can use it to specify that a column value cannot be null. Typically, you apply this constraint when creating a table, as the following example demonstrates:
+
+```sql
+CREATE TABLE employees (
+    employee_no integer NOT NULL,
+    name text NOT NULL,
+    department text
+);
+```
+
+You may also add the `NOT NULL` constraint to one or more columns of an existing table by executing the `ALTER TABLE` statement, as follows:
+
+```sql
+ALTER TABLE employees
+ALTER COLUMN department SET NOT NULL;
+```
+
+## Deferring Constraint Check
+
+YSQL allows you to set constraints using the `SET CONSTRAINTS` statement and defer the contsraints check until the transaction commit time by declaring the contstraint `DEFERRED`, as shown in the following example:
+
+```sql
+BEGIN;
+SET CONSTRAINTS name DEFERRED;
+...
+COMMIT;
+```
+
+
+When creating a foreign key constraint that might need to be deferred (for example, if a transaction could have inconsistent data for a while, such as initially mismatched foreign keys), you have an option to define this transaction as `DEFERRABLE` and `INITIALLY DEFERRED`, as follows:
+
+```sql
+CREATE TABLE employees (
+    employee_no integer,
+    name text,
+    department text
+  		DEFERRABLE INITIALLY DEFERRED
+);
+```
+
+You can change this behavior by altering a constraint for a table or column, as shown in the following example:
+
+```sql
+ALTER TABLE employees 
+ALTER CONSTRAINT employee_no
+  DEFERRABLE INITIALLY DEFERRED;
+```
+
+Note that the `NOT NULL` constraint cannot be used with the `SET CONSTRAINTS` statement.
```

Comment:
This feels like another great place for an example.

Reply:
Thanks.